### PR TITLE
PI Web API Data Analysis using Jupyter Notebook

### PR DESCRIPTION
**Pull request by msingh7910**

### CODE REVIEW

File: piwebapi_samples/Data_Analysis/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	"AF_SERVER_NAME": "AssetServerName",
+	"PI_SERVER_NAME": "PIServerName",
+	"AF_DATABASE_NAME": "AFDatabaseNAme",
+	"USER_NAME": "MyUserName",
+	"USER_PASSWORD": "MyUserPassword",
+	"AUTH_TYPE": "basic"
+```
+- `Building Example.xml` assumes that the PI Server is on the same system. Edit the `Building Example.xml` to replace all occurences of `localhost`to your PI Server.
+- Run `PIUploadUtility.sln` which imports the AF database from `Building Example.xml`, creates PI tags outlined in `tagdefinition.csv` and uploads the values in `pidata.csv` to PI Data Archive.
+- The
```
Comment (derekendres) [inline piwebapi_samples/Data_Analysis/README.md]: Feel like there is more to this sentence... (or maybe less)

File: piwebapi_samples/Data_Analysis/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+
+# PI Web API Data Analysis Sample
+
+**Version:** 1.0.0 
+
+[![Build Status](https://dev.azure.com/osieng/engineering/_apis/build/status/product-readiness/PI-System/PIWebAPI_Data_Analysis?branchName=master)](https://dev.azure.com/osieng/engineering/_build/latest?definitionId=1644&branchName=master)
+
+The sample code in this folder demonstrates how to utilize the PI Web API to do some basic data analysis using Python Jupyter Notebook. In order to run this sample, you need to have [Python](https://www.python.org/downloads/release/python-373/) installed.
```
Comment (derekendres) [inline piwebapi_samples/Data_Analysis/README.md:8]: Do we need a Jupyter download too?

File: piwebapi_samples/Data_Analysis/README.md (hunk 1 above)
Comment (msingh7910) [reply piwebapi_samples/Data_Analysis/README.md]: I don't understand, would you like it to be divided into multiple sentences?

**Review by derekendres — APPROVED** [review]

File: piwebapi_samples/Data_Analysis/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+
```
Comment (dcbroad3) [inline piwebapi_samples/Data_Analysis/README.md]: Extra line of whitespace at the beginning of this file

File: piwebapi_samples/Data_Analysis/README.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+# PI Web API Data Analysis Sample
+
+**Version:** 1.0.0 
+
+[![Build Status](https://dev.azure.com/osieng/engineering/_apis/build/status/product-readiness/PI-System/PIWebAPI_Data_Analysis?branchName=master)](https://dev.azure.com/osieng/engineering/_build/latest?definitionId=1644&branchName=master)
+
+The sample code in this folder demonstrates how to utilize the PI Web API to do some basic data analysis using Python Jupyter Notebook. In order to run this sample, you need to have [Python](https://www.python.org/downloads/release/python-373/) installed.
```
Comment (dcbroad3) [inline piwebapi_samples/Data_Analysis/README.md]: Is there a reason we're linking to a specific version of Python here?

File: piwebapi_samples/Data_Analysis/README.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+## Background and Problem
+### Background
+San Leandro Technical Campus (SLTC) is OSIsoft's headquarters in San Leandro, California. Like every office building, SLTC also needs temperature control systems to maintain a comfortable working environment. Now, consider a normal working day: The building starts getting occupied at around 7 AM. The building management system has to turn on the systems beforehand in order to reach a set temperature (say 70 deg F) by 7 AM. So let's say they have the control systems set to turn on at around 5 AM.
```
Comment (dcbroad3) [inline piwebapi_samples/Data_Analysis/README.md]: I don't think the first `also` is necessary in this first sentence.

File: piwebapi_samples/Data_Analysis/README.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+Our objective here is to predict the total time taken by each of the unit in the building to reach the setpoint so that building management systems can turn on the units as close as possible to 7 AM.
+
+In our earlier example, if we had predicted earlier (to some degree of accuracy) that the Unit A takes an hour to reach the set point, we would have turned it on by 6 AM instead of 5 AM which would have resulted us in saving that extra hour of wasted energy.
+
+We will be utilizing the data that we have and do some basic data analysis and predictive machine learning using Jupyter Notebook in order to predict the cooling time for different units.
+
+### Data Overview
+In this sample, we are denoting these control systems as two different `VAVCO` units which are represented as AF Elements. Each of these elements have different attributes which mimic an actual cooling unit such as `% cooling` and `Set Point Offset`.
+
+There are also eventframes of the template `VAVCO startup` which represent the start up events of the cooling unit. They have some attributes which capture the metadata on start time as well as end time.
```
Comment (dcbroad3) [inline piwebapi_samples/Data_Analysis/README.md]: If we capitalized `AF Elements` we should probably say `Event Frames` or `AF Event Frames`

File: piwebapi_samples/Data_Analysis/UploadUtility/Program.cs [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.InnerException.Message);
+                }
+            }
+        }
+        static void DeleteExistingDatabase(string assetserver)
+        {
+            string databaseName = config["AF_DATABASE_NAME"].ToString();
+            string databasePath = "\\\\" + assetserver + "\\" + databaseName;
```
Comment (dcbroad3) [inline piwebapi_samples/Data_Analysis/UploadUtility/Program.cs]: These are ok, but if you're not familiar with it, you can also do `$"\\\\{assetServer}\\{databaseName}"` without doing the string concatenations.

File: piwebapi_samples/Data_Analysis/UploadUtility/Program.cs [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+using System.IO;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace UploadUtility
+{
+    class Program
+    {
+        static readonly string defaultConfigFile = @"..\..\..\..\test_config.json";
```
Comment (dcbroad3) [inline piwebapi_samples/Data_Analysis/UploadUtility/Program.cs]: Remind me, why are these files expected to be so much higher in the folder hierarchy?

File: piwebapi_samples/Data_Analysis/UploadUtility/Program.cs (hunk 7 above)
Comment (msingh7910) [reply piwebapi_samples/Data_Analysis/UploadUtility/Program.cs]: yeah I guess I can change it to that. It slipped my mind

File: piwebapi_samples/Data_Analysis/UploadUtility/Program.cs (hunk 8 above)
Comment (msingh7910) [reply piwebapi_samples/Data_Analysis/UploadUtility/Program.cs]: the jupyter notebook also uses this test_config file which is at the root.

File: piwebapi_samples/Data_Analysis/README.md (hunk 6 above)
Comment (msingh7910) [reply piwebapi_samples/Data_Analysis/README.md]: sure that makes sense. I'll make the edit.

File: piwebapi_samples/Data_Analysis/README.md (hunk 5 above)
Comment (msingh7910) [reply piwebapi_samples/Data_Analysis/README.md]: Man English is hard. You're right though. I'll push an update.

File: piwebapi_samples/Data_Analysis/README.md (hunk 4 above)
Comment (msingh7910) [reply piwebapi_samples/Data_Analysis/README.md]: No there's no reason. I had copied these from the Python sample of PI Web API

Comment (dcbroad3) [reply piwebapi_samples/Data_Analysis/README.md]: I'd prefer we link to just https://www.python.org/downloads/

File: piwebapi_samples/Data_Analysis/UploadUtility/Program.cs (hunk 8 above)
Comment (msingh7910) [reply piwebapi_samples/Data_Analysis/UploadUtility/Program.cs]: fixed

File: piwebapi_samples/Data_Analysis/README.md (hunk 4 above)
Comment (msingh7910) [reply piwebapi_samples/Data_Analysis/README.md]: fixed

**Pull request merged**